packages/react-aria-components/src/Checkbox.tsx

```diff
@@ -118,8 +119,12 @@ function CheckboxGroup(props: CheckboxGroupProps, ref: ForwardedRef<HTMLDivEleme
     defaultClassName: 'react-aria-CheckboxGroup'
   });
 
+  let DOMProps = filterDOMProps(props);
+  delete DOMProps.id;
```

Comment:
has `delete DOMProps.id` to keep the behavior the same as when useRenderProps deleted the id.

Reply:
Actually, this doesn't need the filterDOMProps since useCheckboxGroup will handle adding it to group props. In fact, the previous implementation of useRenderProps was actually allowing too much through since many of the hooks would apply the top level provided data-testids and apply them to the inner input/group elements. Will go through and update accordingly